### ISSUE TITLE
[snapshot] reset_simulators: Can't reset 11.0 sims using Xcode 9 environment

### ISSUE THREAD

**Issue by sarunw** [fastlane/fastlane#10174]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
`fastlane snapshot reset_simulators --ios 11.0 --verbose` and get 
```
INFO [2017-08-29 11:52:07.32]: $ xcrun simctl list devices
INFO [2017-08-29 11:52:07.40]: ▸ == Devices ==
INFO [2017-08-29 11:52:07.40]: ▸ -- iOS 7.1 --
INFO [2017-08-29 11:52:07.40]: ▸ -- iOS 8.1 --
INFO [2017-08-29 11:52:07.40]: ▸ -- iOS 8.4 --
INFO [2017-08-29 11:52:07.40]: ▸ -- iOS 9.0 --
INFO [2017-08-29 11:52:07.40]: ▸ -- iOS 9.3 --
INFO [2017-08-29 11:52:07.40]: ▸ -- iOS 10.3 --
INFO [2017-08-29 11:52:07.40]: ▸ -- iOS 11.0 --
INFO [2017-08-29 11:52:07.40]: ▸ -- tvOS 11.0 --
INFO [2017-08-29 11:52:07.40]: ▸ -- watchOS 4.0 --
INFO [2017-08-29 11:52:07.55]: $ xcrun simctl list devices
INFO [2017-08-29 11:52:07.61]: ▸ == Devices ==
INFO [2017-08-29 11:52:07.61]: ▸ -- iOS 7.1 --
INFO [2017-08-29 11:52:07.61]: ▸ -- iOS 8.1 --
INFO [2017-08-29 11:52:07.61]: ▸ -- iOS 8.4 --
INFO [2017-08-29 11:52:07.61]: ▸ -- iOS 9.0 --
INFO [2017-08-29 11:52:07.61]: ▸ -- iOS 9.3 --
INFO [2017-08-29 11:52:07.61]: ▸ -- iOS 10.3 --
INFO [2017-08-29 11:52:07.61]: ▸ -- iOS 11.0 --
INFO [2017-08-29 11:52:07.61]: ▸ -- tvOS 11.0 --
INFO [2017-08-29 11:52:07.61]: ▸ -- watchOS 4.0 --
```
Simulator got deleted, but not create.

`xcode-select -p` and got
`/Applications/Xcode-beta.app/Contents/Developer`

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 10.12.6                                          |
| Ruby                        | 2.3.4                                            |
| Bundler?                    | false                                            |
| Git                         | git version 2.8.1                                |
| Installation Source         | ~/.rbenv/versions/2.3.4/bin/fastlane             |
| Host                        | Mac OS X 10.12.6 (16G29)                         |
| Ruby Lib Dir                | ~/.rbenv/versions/2.3.4/lib                      |
| OpenSSL Version             | OpenSSL 1.0.2k  26 Jan 2017                      |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode-beta.app/Contents/Developer/ |
| Xcode Version               | 9.0                                              |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.54.2"

default_platform :ios

platform :ios do
  before_all do

    xcode_select "/Applications/Xcode-9.0.app"
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods unless Helper.is_ci? # circle ci already run pod install, and in circleci currently have problem with this command

    if Helper.is_ci?
      increment_build_number(build_number: ENV['CIRCLE_BUILD_NUM'])
    end
    
    ENV['KEYCHAIN_NAME'] = "TempKeychain.keychain"
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    # snapshot
    gym # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  desc "staging for uat"
  lane :staging do
    import_certificates
    sigh

    # use the UDID of the newly created provisioning profile
    ENV["PROFILE_UUID"] = lane_context[SharedValues::SIGH_UDID]

    gym(scheme: "Rocky Staging")
    pilot(skip_submission: true, skip_waiting_for_build_processing: true)
  end

  #
  # Learn more here: https://github.com/fastlane/setups/blob/master/samples-ios/distribute-beta-build.md 🚀
  #
  lane :beta do |values|
    import_certificates
    sigh(adhoc: true)

    # use the UDID of the newly created provisioning profile
    ENV["PROFILE_UUID"] = lane_context[SharedValues::SIGH_UDID]

    # # Fabric generated this lane for deployment to Crashlytics Beta
    # # set 'export_method' to 'ad-hoc' if your Crashlytics Beta distribution uses ad-hoc provisioning
    gym(scheme: 'Rocky Staging', export_method: 'ad-hoc')

    crashlytics(
      crashlytics_path: "./Pods/Crashlytics",
      api_token: ENV["FABRIC_API_TOKEN"],
      build_secret: ENV["FABRIC_BUILD_SECRET"],
      notifications: true,
      groups: 'testers'
    )
  end

  desc "integration for internal test"
  lane :integration do
    import_certificates
    sigh

    # use the UDID of the newly created provisioning profile
    ENV["PROFILE_UUID"] = lane_context[SharedValues::SIGH_UDID]

    gym(scheme: "Rocky Integration")
    pilot(skip_submission: true, skip_waiting_for_build_processing: true)
  end

  desc "production"
  lane :production do      
    import_production_certificates
    sigh

    # use the UDID of the newly created provisioning profile
    ENV["PROFILE_UUID"] = lane_context[SharedValues::SIGH_UDID]

    gym(scheme: "Rocky Production")
    pilot(skip_submission: true, skip_waiting_for_build_processing: true)  
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end

  def import_certificates
    return unless Helper.is_ci? # As in my local machine I don't need to import the certificates, only run if it's in the CI

    create_keychain(
      name: ENV["KEYCHAIN_NAME"],
      default_keychain: true,
      unlock: true,
      timeout: 3600,
      lock_when_sleeps: true,
      password: ENV["KEYCHAIN_PASSWORD"]
    )

    import_certificate(
      certificate_path: "fastlane/certificates/distribution.p12",
      certificate_password: ENV['CERT_PASSWORD'],
      keychain_name: ENV["KEYCHAIN_NAME"]
    )

    import_certificate(
      certificate_path: "fastlane/certificates/ios_distribution.cer",
      keychain_name: ENV["KEYCHAIN_NAME"]
    )

    # Sierra changes
    Helper.backticks("security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k #{ENV["KEYCHAIN_PASSWORD"]} #{ENV['KEYCHAIN_NAME']}")    
  end

  def import_production_certificates
    return unless Helper.is_ci? # As in my local machine I don't need to import the certificates, only run if it's in the CI

    create_keychain(
      name: ENV["KEYCHAIN_NAME"],
      default_keychain: true,
      unlock: true,
      timeout: 3600,
      lock_when_sleeps: true,
      password: ENV["KEYCHAIN_PASSWORD"]
    )

    import_certificate(
      certificate_path: "fastlane/certificates/production_distribution.p12",
      certificate_password: ENV['PRODUCTION_CERT_PASSWORD'],
      keychain_name: ENV["EYCHAIN_NAME"]
    )

    import_certificate(
      certificate_path: "fastlane/certificates/production_distribution.cer",
      keychain_name: ENV["KEYCHAIN_NAME"]
    )

    # Sierra changes
    Helper.backticks("security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k #{ENV["KEYCHAIN_PASSWORD"]} #{ENV['KEYCHAIN_NAME']}")
  end

end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# app_identifier "com.oozou.rocky" # The bundle identifier of your app
# apple_id "" # Your Apple email address

# You can uncomment any of the lines below and add your own
# team selection in case you're in multiple teams
team_name "Oozou Limited"
team_id "BA36Y82DMA"

# To select a team for iTunes Connect use
itc_team_name "Oozou Limited"
itc_team_id "16134"

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

for_lane :staging do
  app_identifier "com.oozou.rocky.staging"
end

for_lane :beta do
  app_identifier "com.oozou.rocky"
end

for_lane :production do
  team_name "LINE COMPANY (THAILAND) COMPANY LIMITED"
  team_id "W5H4R7GVBZ"

  itc_team_name "LINE COMPANY (THAILAND) COMPANY LIMITED"
  itc_team_id "117742454"

  app_identifier "com.linemobile.th"
end

for_lane :integration do
  app_identifier "com.oozou.rocky.integration"
end```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.54.3  | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.3.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.1        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
| json                      | 2.1.0        |
| io-console                | 0.4.5        |
| excon                     | 0.58.0       |
| faraday                   | 0.13.1       |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.3        |
| faraday_middleware        | 0.12.2       |
| bundler                   | 1.15.3       |
| unicode-display_width     | 1.3.0        |
</details>


*generated on:* **2017-08-29**
</details>

**Comment by casz** [comment]
Same issue.

Even with `fastlane snapshot reset_simulators --ios 10.3` it stills deletes my ios 11 simulators 👎

**Comment by jozsefmihalicza** [comment]
The root cause is that the format of the output of `xcrun simctl list runtimes` has changed from (8.3.2)
`iOS 10.3 (10.3 - 14E269) (com.apple.CoreSimulator.SimRuntime.iOS-10-3)`
to (9.0)
`iOS 11.0 (11.0 - 15A5361a) - com.apple.CoreSimulator.SimRuntime.iOS-11-0

and the regexp in https://github.com/fastlane/fastlane/blob/e93b397900bbea5aa626ea2c44b45c172e444d1c/snapshot/lib/snapshot/reset_simulators.rb#L24 does not find the runtimes for 11.0.

**Comment by casz** [comment]
I think the work already started for using the json format for simctl #9610
Though it would properly be faster to fix the regex.

**Comment by ohayon** [comment]
Yup! This is something that is definitely on our radar! We would be happy to review a PR if anyone wants to make the fix before we get to it! 🚀

**Comment by casz** [comment]
I believe I have fixed the issue with #10215 👍 with https://github.com/fastlane/fastlane/pull/10215/commits/9a1e001cb92e81d88a278abab7fa0d575388bd7a
